gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -113,32 +109,63 @@ impl GatewayLdkClient {
         })?;
 
         let (htlc_stream_sender, htlc_stream_receiver) = tokio::sync::mpsc::channel(1024);
+        let task_group = TaskGroup::new();
+
+        // LDK Node will sync the wallet's chain automatically in the background, but in
+        // the background it only sync every 10 seconds. In devimint we want
+        // this to sync faster, so we spawn a thread that manually syncs the chain.
+        if is_env_var_set(FM_IN_DEVIMINT_ENV) {
+            let node_clone = node.clone();
+            task_group.spawn("ldk node sync event handler", |handle| async move {
+                loop {
+                    if handle.is_shutting_down() {
+                        return;
+                    }
+
+                    if let Err(e) = node_clone.sync_wallets() {
+                        error!(?e, "Failed to sync LDK Node to chain");
+                    }
+
+                    sleep(Duration::from_millis(500)).await;
+                }
+            });
+        }
 
         let node_clone = node.clone();
-        let event_handler_task_handle = spawn("ldk lightning node event handler", async move {
+        task_group.spawn("ldk lightning node event handler", |handle| async move {
             loop {
-                Self::handle_next_event(&node_clone, &htlc_stream_sender).await;
+                Self::handle_next_event(&node_clone, &htlc_stream_sender, &handle).await;
             }
         });
 
         Ok(GatewayLdkClient {
             node,
             esplora_client: esplora_client::Builder::new(esplora_server_url).build_async()?,
-            event_handler_task_handle,
+            task_group,
             htlc_stream_receiver_or: Some(htlc_stream_receiver),
         })
     }
 
     async fn handle_next_event(
         node: &ldk_node::Node,
         htlc_stream_sender: &Sender<Result<InterceptHtlcRequest, Status>>,
+        handle: &TaskHandle,
     ) {
+        let event = tokio::select! {
+            event = node.next_event_async() => {
+                event
+            }
+            () = handle.make_shutdown_rx() => {
+                return;
+            }
+        };
```

Comment:
Nit: could you leave a comment please why the manual select over a shutdown receiver is used here over cancelling the entire future in case of shutdown? 

My read is that we *don't* want to shut down after receiving an even and before having sent it over the HTLC stream sender? Likely related to safe shutdowns? 

But I could imagine being like "that's silly, let's move the shutdown a level up" when looking at the code again (indeed that was my first reaction this time).

Reply:
Yes, your read is correct, if we receive a payment we ideally want to process it before shutting down. This is the pattern we use in LND as well. I will add a comment